Copy edit custom component documentation (#8547)
open-telemetry/opentelemetry.io: content/en/docs/collector/building/receiver.md

> Traces that track the progression of a single request, known as a trace, as it
> is handled by services that make up an application. The request may be
> initiated by a user or an application. Distributed tracing is a form of
> tracing that traverses process, network, and security boundaries.

Although distributed traces are defined in an application-centric way, you can
think of them as a timeline for _any_ request that moves through your system.
Each distributed trace shows how long a request took from start to finish and
breaks down the steps taken to complete it.

If your system generates tracing telemetry, you can configure your
[OpenTelemetry Collector](/docs/collector/) with a trace receiver designed to
receive and convert that telemetry. The receiver converts your data from its
original format into the OpenTelemetry trace model so the Collector can process
it.

To implement a trace receiver, you need the following:

- A `Config` implementation so the trace receiver can gather and validate its
  configurations in the Collector config.yaml.

- A `receiver.Factory` implementation so the Collector can properly instantiate
  the trace receiver component.

> - A `TracesReceiver` implementation that collects the telemetry, converts it to

Reviewer: Thanks! I updated it.

{{% alert title="Check your work" %}}

> - Updated the `Start()` method by initializing the `host` field with the

Reviewer: ```suggestion
Updated the `Start()` method by adding the initialization to the `host` field with the
```

{{% alert title="Check your work" %}}

- Imported the `go.opentelemetry.io/collector/consumer` which is where the
  pipeline's consumer types and interfaces are declared.
- Imported the `go.uber.org/zap` package, which is what the Collector uses for
  its debugging capabilities.
- Added a `zap.Logger` field named `logger` so we can have access to the
  Collector logger reference from within the receiver.
- Added a `consumer.Traces` field named `nextConsumer` so we can push the traces
  generated by the `tailtracer` receiver to the next consumer, declared in the
  Collector pipeline.
- Added a `Config` field named `config` so we can have access to receiver's
  configuration settings defined in the Collector config.

> - Added a variable named `interval`, initialized as a `time.Duration` based on

Reviewer: ```suggestion
- Added a variable named `interval` that is initialized as a `time.Duration` based on
```

{{% alert title="Check your work" %}}

- Imported the receiver module

>   `github.com/open-telemetry/opentelemetry-tutorials/trace-receiver/tailtracer`

Reviewer: ```suggestion
  `github.com/open-telemetry/opentelemetry-tutorials/trace-receiver/tailtracer`,
```

>   , which is where the receiver types and functions are.

Reviewer: ```suggestion
  which is where the receiver types and functions are.
```

  `pcommon.Map` reference returned by the `resource.Attributes()` call.
- Used the `PutInt()` and `PutStr()` methods from `pcommon.Map` to add int and
  string attributes based on the equivalent `Atm` field types. Notice that
  because these attributes are specific to the `Atm` entity, they are all

>   grouped under the "atm." prefix.

Reviewer: ```suggestion
  because these attributes are specific to and only represent the `Atm`
  entity, they are all grouped within the `atm.` prefix.
```

```

At this point, you have everything needed to represent the telemetry generation
entities in your system, as well as the instrumentation responsible for
identifying operations and generating the traces for the system. The next step
is to create the spans representing the operations that the given
instrumentation scope generated as part of a trace.

`ptrace.ScopeSpans` has a method named `Spans()` which returns an instance of a
helper type called `ptrace.SpanSlice`. The `ptrace.SpanSlice` type has methods
to help you handle the array of `ptrace.Span`. The array will contain as many
items as the number of operations the instrumentation scope was able to identify
and describe as part of the trace.

`ptrace.SpanSlice` has a method named `AppendEmpty()` that adds a new
`ptrace.Span` to the array and return its reference.

`ptrace.Span` has the following methods to describe an operation:

> - `SetTraceID(v pcommon.TraceID)` sets the `pcommon.TraceID` by uniquely

Reviewer: ```suggestion
- `SetTraceID(v pcommon.TraceID)` sets the `pcommon.TraceID` uniquely
```

```

You now have the `BackendSystem` entity and its operations represented in spans

> in a proper trace context! Next, you need to push the generated trace via the

Reviewer: ```suggestion
in a proper trace context! Next, you need to push the generated trace through the
```
